src/System.Security.Cryptography.Pkcs/ref/System.Security.Cryptography.Pkcs.netcoreapp.cs

```diff
@@ -5,8 +5,72 @@
+        public Rfc3161TimestampToken AcceptResponse(ReadOnlyMemory<byte> source, out int bytesRead) => throw null;
+        public bool TryAcceptResponse(ReadOnlyMemory<byte> source, out int bytesRead, out Rfc3161RequestResponseStatus status, out Rfc3161TimestampToken token) => throw null;
+        public static Rfc3161TimestampRequest BuildForData(ReadOnlySpan<byte> data, HashAlgorithmName hashAlgorithm, Oid requestedPolicyId = null, ReadOnlyMemory<byte>? nonce = null, bool requestSignerCertificates = false, X509ExtensionCollection extensions = null) => throw null;
+        public static Rfc3161TimestampRequest BuildForHash(ReadOnlyMemory<byte> hash, HashAlgorithmName hashAlgorithm, Oid requestedPolicyId = null, ReadOnlyMemory<byte>? nonce = null, bool requestSignerCertificates = false, X509ExtensionCollection extensions = null) => throw null;
+        public static Rfc3161TimestampRequest BuildForHash(ReadOnlyMemory<byte> hash, Oid hashAlgorithmId, Oid requestedPolicyId = null, ReadOnlyMemory<byte>? nonce = null, bool requestSignerCertificates = false, X509ExtensionCollection extensions = null) => throw null;
+        public static Rfc3161TimestampRequest BuildForSignerInfo(SignerInfo signerInfo, HashAlgorithmName hashAlgorithm, Oid requestedPolicyId = null, ReadOnlyMemory<byte>? nonce = null, bool requestSignerCertificates = false, X509ExtensionCollection extensions = null) => throw null;
+        public static bool TryParse(ReadOnlyMemory<byte> source, out int bytesRead, out Rfc3161TimestampRequest request) => throw null;
+    }
+    public sealed partial class Rfc3161TimestampToken
+    {
+        public Rfc3161TimestampTokenInfo TokenInfo => throw null;
+        public SignedCms AsSignedCms() => throw null;
+        public bool VerifyData(ReadOnlySpan<byte> data) => throw null;
+        public bool VerifyHash(ReadOnlySpan<byte> hash) => throw null;
+        public bool CheckCertificate(X509Certificate2 tsaCertificate) => throw null;
+        public static bool TryParse(ReadOnlyMemory<byte> source, out int bytesRead, out Rfc3161TimestampToken token) => throw null;
+    }
+    public sealed partial class Rfc3161TimestampTokenInfo : AsnEncodedData
+    {
+        public Rfc3161TimestampTokenInfo(byte[] timestampTokenInfo) { }
+        public Rfc3161TimestampTokenInfo(Oid policyId, Oid hashAlgorithmId, ReadOnlyMemory<byte> messageHash, ReadOnlyMemory<byte> serialNumber, DateTimeOffset timestamp, long? accuracyInMicroseconds=null, bool isOrdering=false, ReadOnlyMemory<byte>? nonce=null, ReadOnlyMemory<byte>? tsaName=null, X509ExtensionCollection extensions =null) { throw null; }
+        public int Version => throw null;
+        public Oid PolicyId=> throw null;
+        public Oid HashAlgorithmId => throw null;
+        public ReadOnlyMemory<byte> GetMessageHash() { throw null; }
+        public ReadOnlyMemory<byte> GetSerialNumber() { throw null; }
+        public DateTimeOffset Timestamp => throw null;
```

Comment:
nit: Having `TimeStamp` inside of `TimestampTokenInfo` which is inside of `TimstampToken` feels a bit timestamp-y. Why not call it `GeneralizedTime`?

Reply:
or `CreationTime`?

Reply:
`GeneralizedTime` is the name of the encoding format, it doesn't make sense as an API.  Windows' version is `ftTime`: https://msdn.microsoft.com/en-us/library/windows/desktop/dd433838(v=vs.85).aspx

Reply:
`CreationTime` is reasonable:

> `genTime` is the time at which the time-stamp token has been created by the TSA.

(From RFC 3161)